### ISSUE TITLE
NumericTextBoxComponent: Setting format property through directive does not work initially 

### ISSUE THREAD

**Issue by dvdgutzmann** [telerik/kendo-angular#1903]
## I'm submitting a...

* Bug report 

## Current behavior

We have a directive through which we want to set certain properties of NumericTextBoxComponent. Some are picked up initially (e.g. the spinners property), but setting the format property does not work initially. Only after first clicking into the NumericTextBox and then somewhere else on the page, the format set through the directive seems to be recognized.

## Expected behavior

The directive should be able to set all properties of the NumericTextBox and the changed properties should take effect immediately.

## Minimal reproduction of the problem with instructions
https://stackblitz.com/edit/kendo-angukar-numerictextbox-with-numericbehaviordirective

## Environment

Package version and System:
Freshly forked from the Basic usage example https://www.telerik.com/kendo-angular-ui/components/inputs/numerictextbox/

Browser:

- Chrome Version 70.0.3538.77 (Official Build) (64-bit)

**Comment by danielkaradachki** [comment]
Should work if you set the options in the constructor:
https://stackblitz.com/edit/kendo-angukar-numerictextbox-with-numericbehaviordirecti-nwm7xc?file=app/numericBehavior.directive.ts

The value is already set by the numerictextbox in the OnInit handler of the directive. We will consider exposing a method that updates the input value if this becomes a common issue.

**Comment by dvdgutzmann** [comment]
I might have oversimplified the repro. In our use case we are depending on the ability to set the format property in the ngOnInit handler of the directive. Please see the updated repro: https://stackblitz.com/edit/kendo-angular-numerictextbox-and-directive-with-input . In our production  code the actual values for the format are of course not hardcoded but derived from other values.

Thank you for your time.

**Comment by dvdgutzmann** [comment]
@danielkaradachki colleagues of mine are pretty sure this used to work and should, therefore, be considered a regression.  Give me a heads up if you need me to investigate further.

**Comment by danielkaradachki** [comment]
Not sure how would this have worked. The component value was always updated in the OnChanges hook which is fired before the directive OnInit handler. 
In general, setting directly the component properties is not supported because the Angular change detection works only through template bindings, but we will expose a method that can be used to update the current input value. For now you can use the ControlValueAccessor writeValue method:

https://stackblitz.com/edit/kendo-angular-numerictextbox-and-directive-with-input-wlk4a7?file=app/numericBehavior.directive.ts

**Comment by dvdgutzmann** [comment]
What would you suggest as a proper solution instead of trying to wrap "common behavior" in a directive? Is there a more "angularly way" which would also work with Kendo components? Any hint is appreciated!

The workaround using .writeValue works, however, thank you!

**Comment by danielkaradachki** [comment]
A possible approach is to wrap the component and set its properties to through the template.

**Comment by dvdgutzmann** [comment]
We have quite a few custom components, all wrapping NumericTextBox (and also other components). By utilizing the directive we tried to wrap the "common behavior" part (format, turned off spinners, ...) because in our use case that has to be consistent across all the custom components where we used NumericTextBox. We'll revisit this design decision in the future and, in the meantime, look forward to the enhancement.
